File: basics/Q56/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var (
		x1 float64
		x2 float64
		y1 float64
		y2 float64
	)
	fmt.Print("Enter x1: ")
	fmt.Scan(&x1)
	fmt.Print("Enter x2: ")
	fmt.Scan(&x2)
	fmt.Print("Enter y1: ")
	fmt.Scan(&y1)
	fmt.Print("Enter y2: ")
	fmt.Scan(&y2)
	dist := math.Sqrt(math.Pow((x2-x1), 2) + math.Pow((y2-y1), 2))
	fmt.Println("Distance is:", dist)
}

File: basics/Q56/main.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter x1:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let x1: f64 = input.trim().parse().expect("Not a valid number");

    input.clear();
    println!("Enter x2:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let x2: f64 = input.trim().parse().expect("Not a valid number");

    input.clear();
    println!("Enter y1:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let y1: f64 = input.trim().parse().expect("Not a valid number");

    input.clear();
    println!("Enter y2:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let y2: f64 = input.trim().parse().expect("Not a valid number");

    let dist = ((x2 - x1).powi(2) + (y2 - y1).powi(2)).sqrt();
    println!("Distance is: {}", dist);
}